### DISCUSSION TITLE
이 문제 어떻게 접근해야 할까요???

### DISCUSSION THREAD

**Discussion by dmdmk1** [babiesdev/algorithm-study#22] [Q&A]
도형 문제는 정말 막막하네요 ㅠㅠ
**_퍼즐 조각 채우기_** - https://programmers.co.kr/learn/courses/30/lessons/84021
<img width="524" alt="스크린샷 2021-10-17 오후 3 22 24" src="https://user-images.githubusercontent.com/61447654/137614313-3a0b78ca-abdc-409a-af33-f8060bbaaa76.png">

**Comment by gunkim** [comment]
음.. 정말 그러네요... 저도 한번 같이 풀어보겠습니다!

**Comment by coloredrabbit** [comment]
전 이렇게 풀엇습니다.
1. game_board에서 비어있는 부분의 좌표를 모두 가져옵니다. (단 크기가 7이상이면 제외)
2. table에서 채워져 있는 부분의 좌표를 모두 가져옵니다.
3. 1.과 2.에서 위치에 대해 매칭이 되는 경우를 확인합니다.
  3-1. 만약 매칭이 된다면 답에 바로 포함합니다.
  3-2. 매칭이 되지 않았다면 다음으로 넘어갑니다.

매칭 알고리즘은 다음과 같이 짰습니다.
1. 도형의 하나 위치를 선정하여 원점 O(0, 0)으로 설정합니다.
2. 도형을 구성하는 다른 위치를 이 원점을 기준으로 상대 좌표를 계산합니다.
3. game_board에서 빈칸형태 중에서 하나를 A라고하고 A를 구성하는 집합을 다음과 같이 표현한다면 A={(x1, y1), (x2, y2, (x3, y3)}
    3-1. (x1, y1)을 원점에 두고 90도씩 회전하면서 매칭(모든 좌표가 정확히 일치함)되는지 봅니다.
    3-2. (x2, y2)을 원점에 두고 90도씩 회전하면서 매칭되는지 봅니다.
    3-3. (x3, y3)을 원점에 두고 90도씩 회전하면서 매칭되는지 봅니다.
    (참고로 (xn, yn)을 원점에 둔다면 A의 구성하는 모든 좌표 역시 상대적인 좌표로 구해야합니다.
